### ISSUE TITLE
Modula lexer bug

### ISSUE THREAD

**Issue by 1997tn** [ScintillaOrg/lexilla#129]
MOD is a keyword for modulo on both Modula-2 and Modula-3. The problem is, if MOD is part of another word, e.g: HMOD, it will also highlight MOD like this: H**MOD**. It should only highlight MOD when it's a standalone word!

**Comment by nyamatongwe** [comment]
Please add an example file that illustrates the problem.

**Comment by 1997tn** [comment]
> Please add an example file that illustrates the problem.

No example file needed. Open a new file on any Scintilla based editor (e.g: BowPad, it uses Lexilla). Set the lexer to Modula. Type HMOD into it, you will immediately see it's highlighted as H**MOD**. Only MOD as a standalone word should be highlighted.

**Comment by nyamatongwe** [comment]
An example file is necessary to reproduce and fix the problem and then ensure that the fix is checked against future changes. Moving the onus of producing an example file onto bug fixers makes it more difficult to reproduce the bug and less likely that the bug will be addressed at all.

**Comment by 1997tn** [comment]
> An example file is necessary to reproduce and fix the problem and then ensure that the fix is checked against future changes. Moving the onus of producing an example file onto bug fixers makes it more difficult to reproduce the bug and less likely that the bug will be addressed at all.

If you so wanted it:

```
INTERFACE Test;

TYPE
  (* Opaque types *)
  HANDLE    = ADDRESS;
  HMOD      = HANDLE;

END Test.
```

As I told you, just open a Modula-3 source file (.i3 or .m3) and type HMOD, you will see it's highlighted as H**MOD**. It's so simple like that, why a sample file is even needed? Always asking for a sample file is an overly too rigid discipline that really doesn't needed at all in this situation :-1:

**Comment by rdipardo** [comment]
This appears to be a trivial fix.

[Issue129.zip](https://github.com/ScintillaOrg/lexilla/files/10472207/Issue129.zip)

The keyword matching loop beginning at line 365 has an empty `else` branch that should be advancing the position indicator by the length of a whole word:

https://github.com/ScintillaOrg/lexilla/blob/557d750299f2c27b0f35e219e686602eb5f6ff43/lexers/LexModula.cxx#L387-L389

The author's memo here suggests he would've corrected the logic himself given more time.

**Comment by nyamatongwe** [comment]
@rdipardo,

The patch appears to `Forward` 1 too many times as there is a `Forward` at the loop end. For example, the comment immediately following in this example doesn't get comment style:
~~~Modula-3
  HMOD(* Handle of module *)       = HANDLE;
~~~
That doesn't occur just after keywords.

**Comment by rdipardo** [comment]
> The patch appears to `Forward` 1 too many times as there is a `Forward` at the loop end.

Preventing an early exit with `continue` seems to keep the final position within bounds.
This updated set also amends the test case by adding an inline comment.

[Issue129-iter-2.zip](https://github.com/ScintillaOrg/lexilla/files/10473402/Issue129-iter-2.zip)

**Comment by nyamatongwe** [comment]
Fixed in 5.2.2.
